apps/payments/next/app/[locale]/[offeringId]/checkout/[interval]/[cartId]/layout.tsx

```diff
@@ -52,9 +53,10 @@ export default async function RootLayout({
   const cartDataPromise = getCartAction(params.cartId);
   const l10n = getApp().getL10n(locale);
   const [cms, cart] = await Promise.all([cmsDataPromise, cartDataPromise]);
+  console.log('layout cart', cart);
 
   return (
-    <>
+    <Template cart={cart}>
```

Comment:
Doesn't Next.js automatically wrap the page with `template`, similar to how it automatically wraps pages with `layout`?

That said, does `template` make sense here? Looking at the Next.js docs, it doesn't seem like the recommended usage for `template` applies here. Happy to discuss further.

> While less common, you might choose to use a template over a layout if you want:
> - Features that rely on useEffect (e.g logging page views) and useState (e.g a per-page feedback form).
> - To change the default framework behavior. For example, Suspense Boundaries inside layouts only show the fallback the first time the Layout is loaded and not when switching pages. For templates, the fallback is shown on each navigation.

Reply:
Just wrapped it in a `useEffect` callback, which is probably more correct anyhow! It seems like the intended use-case of a template to me though -- the need here is to have a layer that watches the `cart` variable, and then makes an update call to Sentry depending on the cart state